### PR TITLE
Added Simulation features common test

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

## Summary

Added Simulation features common test

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by ahcorde** [comment]
@osrf-jenkins run tests please!

File: test/common_test/simulation_features.cc [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+    loader.LoadLib(SimulationFeaturesTest::GetLibToTest());
+
+    // TODO(ahcorde): We should also run the 3f, 2d, and 2f variants of
+    // FindFeatures
+    pluginNames = gz::physics::FindFeatures3d<Features>::From(loader);
+    if (pluginNames.empty())
+    {
+      std::cerr << "No plugins with required features found in "
+                << GetLibToTest() << std::endl;
+      // TODO(ahcorde): If we update gtest we can use here GTEST_SKIP()
```
Comment (mjcarroll) [inline test/common_test/simulation_features.cc:108]: Is GTEST_SKIP not available in the current version?  I can prioritize getting gtest updated.

File: test/common_test/simulation_features.cc [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+                gz::math::eigen3::convert(frameData.linearVelocity), 0.1));
+    EXPECT_EQ(gz::math::Vector3d(0.4, 0.5, 0.6),
+              gz::math::eigen3::convert(frameData.angularVelocity));
+  }
+}
+
+
+TEST_F(SimulationFeaturesTest, CollideBitmasks)
+{
+  auto worlds = LoadWorlds(loader, pluginNames, TEST_WORLD_DIR "/shapes_bitmask.sdf");
```
Comment (mjcarroll) [inline test/common_test/simulation_features.cc:366]: joinPaths

File: test/common_test/simulation_features.cc (hunk 1 above)
Comment (ahcorde) [reply test/common_test/simulation_features.cc:108]: yep, it's not available

File: test/common_test/simulation_features.cc [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+    // Expect box_filtered and box_colliding to collide with box_base
+    contacts = world->GetContactsFromLastStep();
+    EXPECT_NE(0u, contacts.size());
+  }
+}
+
+
+TEST_F(SimulationFeaturesTest, RetrieveContacts)
+{
+  auto worlds = LoadWorlds(loader, pluginNames, TEST_WORLD_DIR "/shapes.world");
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply test/common_test/simulation_features.cc]: https://github.com/gazebosim/gz-physics/pull/362/commits/edf9540da3ecd489812979315bf19c89d1cd448f

File: test/common_test/CMakeLists.txt [hunk 4]
```diff
@@ -3,6 +3,7 @@ set(TEST_TYPE "COMMON_TEST")
 set(tests
   basic_test
   construct_empty_world
+  simulation_features
 )
 
 link_directories(${PROJECT_BINARY_DIR}/test)
```
Comment (mjcarroll) [inline test/common_test/CMakeLists.txt:9]: Not related to this PR, but link_directories probably isn't necessary here?

File: test/common_test/simulation_features.cc [hunk 5]
```diff
@@ -0,0 +1,583 @@
...
+  gz::physics::AttachSphereShapeFeature,
+  gz::physics::AttachCylinderShapeFeature,
+  gz::physics::AttachEllipsoidShapeFeature,
+  gz::physics::AttachCapsuleShapeFeature,
+  gz::physics::GetSphereShapeProperties,
+  gz::physics::GetBoxShapeProperties,
+  gz::physics::GetCylinderShapeProperties,
+  gz::physics::GetCapsuleShapeProperties,
+  gz::physics::GetEllipsoidShapeProperties
+>;
```
Comment (mjcarroll) [inline test/common_test/simulation_features.cc:90]: Can these be grouped into feature lists that can be reused?

**Review by mjcarroll — APPROVED** [review]
Two minor comments, otherwise LGTM

File: test/common_test/CMakeLists.txt (hunk 4 above)
Comment (ahcorde) [reply test/common_test/CMakeLists.txt]: https://github.com/gazebosim/gz-physics/pull/362/commits/693a109fabbc8e73090ab71b9054dfebd443beef

**Pull request merged**